irods/test/meta_test.py

```diff
@@ -520,13 +520,41 @@ def units():
                 if d: d.unlink(force = True)
                 helpers.remove_unused_metadata(session)
 
+    def test_AVUs_populated_improperly_with_empties_or_nonstrings_fail_identically__issue_547(self):
+        try:
+            to_delete = []
+            hc = helpers.home_collection(self.sess)
+            mtemplate = iRODSMeta('some_name', 'some_value', 'some_units')
+            for index in ('name','value','units'):
+                for edge_case_arg in ('', 3):
+                    # Empty units are permitted, ie. iRODSMeta(attr,value,'') and iRODSMeta(attr,value) are equivalent.
+                    if edge_case_arg in ('',b'') and index == 'units':
+                        continue
+                    for method in ('set','add'):
+                        data = self.sess.data_objects.create("{hc}/{index}_{edge_case_arg}_{method}_AZ__issue_547".format(**locals()))
+                        to_delete.append(data)
+                        m = iRODSMeta(*mtemplate)
+                        setattr(m, index, edge_case_arg)
```

Comment:
So these lines effectively copy `mtemplate` and override the value identified by `index` with the value of `edge_case_arg`, correct?

Reply:
Yes, exactly.